gamePlay.js

```diff
@@ -0,0 +1,88 @@
+import { displayRevealedCell, revealWholeGrid, resetDisplay, getCellElem} from "./display.js";
+import Grid from "./Grid.js";
+
+
+export const changeDisplayedCell = function (celElem, cell, gridObject, game) {
+    if (game.gameOver){
+        return;
+    }
+
+    if (game.isFlaggingOn){
+        toggleFlagging(cell, celElem);
+        return;
+    }
+
+    if (cell.numMinesAround == 0){
+        gridObject.revealZeros(cell).forEach(c => displayRevealedCell(c));
+        return;
+    }
+
+    if (cell.isMine){
+        cell.isRevealed = true;
+        revealWholeGrid(gridObject);
+        game.gameOver = true;
+        return;
+    }
+
+    cell.isRevealed = true;
+    displayRevealedCell(cell);
+    return;   
+};
+
+export const clickToReveal = function (gridObject, game){
+    gridObject.cells.forEach(cell => {
+        let cellElem = getCellElem(cell);
+        cellElem.addEventListener("click", () => 
+            changeDisplayedCell(cellElem, cell, gridObject, game)
+        );
+    });
+}
+
+
+
+export const toggleFlagging = function (object, cellElem){
+    object.isFlaggingOn ? cellElem.classList.remove("flag") : cellElem.classList.add("flag");
+    object.isFlaggingOn = !object.isFlaggingOn;
+}
+
+export const activateFlaggingButton = function(game) {
```

Comment:
this function is great - it has exactly the things it should in there and the variables are named very well. it's very clear at a glance what this function is supposed to do. well done

Reply:
Thank you for the positive feedback!!